src/server.rs

```diff
@@ -248,8 +248,12 @@ fn read_opt(
         None => true,
         Some(0) => true,
         Some(_) => {
-            let last = messages.last().unwrap();
-            (last.idx as u64) == max_index
+            if messages.len() == 0 {
+                true
+            } else {
+                let last = messages.last().unwrap();
+                (last.idx as u64) == max_index
+            }
```

Comment:
This whole block should probably be:
```rust
messages.last().map(|last| (last.idx as u64) == max_index).unwrap_or(true)
```

Reply:
Thanks! I keep forgetting that last() is effectively an iterator of one item.

Reply:
I meant the entire if/else block, the test for is_empty isn't needed.